### DISCUSSION TITLE
No connection to inverter. Bad hardware?

### DISCUSSION THREAD

**Discussion by dzecevic** [tbnobody/OpenDTU#271] [Q&A]
Hi!

I am trying to connect, but no way.

Always the same in log:

17:41:03.095 > Fetch inverter: 114182126368

17:41:03.100 > Request SystemConfigPara

17:41:03.102 > TX RealTimeRunData Channel: 23 --> 15 82 12 63 68 98 01 86 08 80 0B 00 63 58 03 8E 00 00 00 00 00 00 00 00 86 D3 F1 

17:41:03.347 > RX Period End

17:41:03.349 > All missing

17:41:03.349 > Nothing received, resend whole request

17:41:03.353 > TX RealTimeRunData Channel: 40 --> 15 82 12 63 68 98 01 86 08 80 0B 00 63 58 03 8E 00 00 00 00 00 00 00 00 86 D3 F1 

17:41:03.595 > RX Period End

17:41:03.598 > All missing

17:41:03.599 > Nothing received, resend whole request

17:41:03.602 > TX RealTimeRunData Channel: 61 --> 15 82 12 63 68 98 01 86 08 80 0B 00 63 58 03 8E 00 00 00 00 00 00 00 00 86 D3 F1 

17:41:03.845 > RX Period End

17:41:03.848 > All missing

17:41:03.848 > Nothing received, resend whole request

Not sure what is wrong. Do i have bad, or non compatible HW?

I got: https://www.makershop.de/module/funk/2x-nrf24l01/
and: https://www.makershop.de/plattformen/nodemcu/espressif-esp32-dev-kit-board/

This looks good:
![image](https://user-images.githubusercontent.com/6962292/197820970-d428a800-6d3d-4248-8f73-9fce7e6534f9.png)

Both leds are on on the board, red and blue.

Not sure where to look fruther.

EDIT I am 1m from Inverter without walls, i tried also few cm from Inverner. And i am using Hoymiles HM-600. I have entered serial correctly.

**Comment by tbnobody** [comment]
The SPI connection to the inverter seems to work correctly (it says "connected" and is able to read the chip type). If you've double checked the IRQ pin and also tripple checked the inverter serial number, it could be that its related to the RF module. We already had several messages that a change of the module fixed the issue. Did you already try the second module you bought?

**Comment by dzecevic** [comment]
Thanks for info.

I found the error, after your info with IRQ pin. It was it. Basicly the pin was correct on board, that was the only way to get the blue light, but then in platformio.ini file i added build flags with pins, and it works.

The IRQ pin on my board is not pin 16, and i do not have pin 16, it is pin 2.
After adding this code to [env:generic]

build_flags = ${env.build_flags}
    -DHOYMILES_PIN_MISO=19
    -DHOYMILES_PIN_MOSI=23
    -DHOYMILES_PIN_SCLK=18
    -DHOYMILES_PIN_IRQ=2
    -DHOYMILES_PIN_CE=4
    -DHOYMILES_PIN_CS=5

I got connected to inverter.
Thanks!

**Reply by madmartin** [reply]
Of course your ESP32 board has a GPIO16, I assume it is labeled with "RX2" or similar. It is usually located near D4.
I recommend to use the default pinout, then you can just download the pre-build firnware later. 

Pinout sample image:
https://i0.wp.com/randomnerdtutorials.com/wp-content/uploads/2018/08/ESP32-DOIT-DEVKIT-V1-Board-Pinout-36-GPIOs-updated.jpg?quality=100&strip=all&ssl=1

**Comment by dzecevic** [comment]
There is RX2.... But for now got it working this way. Also 3.3v is o by me on the other side.

Here is my pin out a connection and it works.
![IMG20221026094628](https://user-images.githubusercontent.com/6962292/197966421-d4024b71-6bb7-4460-922a-565f6224266a.jpg)
![IMG20221026094634](https://user-images.githubusercontent.com/6962292/197966429-5a7bbc72-333a-4d9f-8e52-d0e62a832b36.jpg)
